test/lib/eventFactory.test.js

```diff
@@ -167,6 +205,43 @@ describe('Event Factory', () => {
             })
         );
 
+        it('should create builds if using new workflow', () => {
+            pipelineMock.jobs = Promise.resolve(jobsMock);
+            config.startFrom = '~pr';
+            config.prRef = 'branch';
+            buildFactoryMock.create.resolves(null);
+
+            return factory.create(config).then((model) => {
+                assert.instanceOf(model, Event);
+                assert.calledTwice(buildFactoryMock.create);
+                assert.calledWith(buildFactoryMock.create.firstCall, sinon.match({
+                    eventId: model.id,
+                    jobId: 1,
+                    prRef: 'branch'
+                }));
+                assert.calledWith(buildFactoryMock.create.secondCall, sinon.match({
+                    eventId: model.id,
+                    jobId: 3,
+                    prRef: 'branch'
+                }));
+            });
+        });
+
+        it('should create build if startFrom is a jobName', () => {
+            pipelineMock.jobs = Promise.resolve(jobsMock);
+            config.startFrom = 'component';
```

Comment:
Let's try this test with`integration` job instead of `component`. Incidentally, what will happen if `startFrom` is a disabled job? It doesn't look like you'd get an error (but you should), and we should have a test for this case.

Reply:
Also a test for `startFrom` is a job that doesn't exist.

Reply:
@petey  The expected behavior is reject in both cases?